lib/storage/src/types.rs

```diff
@@ -264,3 +266,25 @@ impl Anonymize for ClusterStatus {
         }
     }
 }
+
+/// Metadata describing extra properties for each peer
+#[derive(Debug, Hash, Serialize, Deserialize, Clone, Eq, PartialEq)]
+pub struct PeerMetadata {
+    /// Peer Qdrant version string
+    version: Option<String>,
```

Comment:
Should we make it three integers instead of a `String`? Or introduce `struct Version { major: u32, minor: u32, patch: u32 }`, maybe?

Reply:
I'm not like super against it, but but how will you implement feature gates then?

Let's say we merged it, and you want to feature-gate something, only if all nodes are at least 1.9.0. We released 1.10.0, how do you implement the comparison? You can't simply compare *strings*, cause "1.10.0" < "1.9.0"

You'll *have to* use `semver::Version` and parse the string every time? :/

Reply:
I'd better use `semver::Version` straight away, cause I don't think that string version is all that useful.